packages/start-slicemachine/src/StartSliceMachineProcess.ts

```diff
@@ -0,0 +1,196 @@
+/* eslint-disable no-console */
+
+import {
+	PrismicUserProfile,
+	SliceMachineManager,
+	createSliceMachineManager,
+} from "@slicemachine/manager";
+import chalk from "chalk";
+import open from "open";
+
+import { createSliceMachineServer } from "./lib/createSliceMachineServer";
+import { listen } from "./lib/listen";
+
+const DEFAULT_SERVER_PORT = 9999;
+
+type CreateStartSliceMachineProcessArgs = ConstructorParameters<
+	typeof StartSliceMachineProcess
+>;
+
+export const createStartSliceMachineProcess = (
+	...args: CreateStartSliceMachineProcessArgs
+): StartSliceMachineProcess => {
+	return new StartSliceMachineProcess(...args);
+};
+
+export type StartSliceMachineProcessConstructorArgs = {
+	open: boolean;
+	port?: number;
+};
+
+/**
+ * Manages the process that runs Slice Machine's server.
+ */
+export class StartSliceMachineProcess {
+	/**
+	 * Determines if Slice Machine should automatically be opened once the server
+	 * starts.
+	 *
+	 * @defaultValue `false`
+	 */
+	open: boolean;
+
+	/**
+	 * The port on which to start the Slice Machine server.
+	 *
+	 * @defaultValue `9999`
+	 */
+	port: number;
+
+	/**
+	 * The Slice Machine manager used for the process.
+	 */
+	private _sliceMachineManager: SliceMachineManager;
+
+	constructor(args: StartSliceMachineProcessConstructorArgs) {
+		this._sliceMachineManager = createSliceMachineManager();
+
+		this.open = args.open ?? false;
+		this.port = args.port ?? DEFAULT_SERVER_PORT;
+	}
+
+	/**
+	 * Runs the process.
+	 */
+	async run(): Promise<void> {
+		await this._sliceMachineManager.plugins.initPlugins();
+
+		await this._validateProject();
+
+		const server = await createSliceMachineServer({
+			sliceMachineManager: this._sliceMachineManager,
+		});
+		const { address } = await listen(server, { port: this.port });
+		const url = `http://localhost:${address.port}`;
+
+		if (this.open) {
+			await open(url);
+		}
+
+		console.log();
+		console.log(
+			await this._buildSliceMachineRunningLine(
+				`Running at ${chalk.magenta(url)}`,
+			),
+		);
+		console.log(this._buildLoggedInAsLine(chalk.dim("Loading...")));
+		console.log();
+
+		const profile = await this._fetchProfile();
+
+		// Non-TTY environments (like GitHub Actions) do not support line management.
+		process.stdout.moveCursor?.(0, -2);
+		process.stdout.clearLine?.(1);
+		console.log(
+			this._buildLoggedInAsLine(
+				profile
+					? `${[profile.firstName, profile.lastName]
+							.filter(Boolean)
+							.join(" ")} ${chalk.dim(`(${profile.email})`)}`
+					: chalk.dim("Not logged in"),
+			),
+		);
+		console.log();
+
+		// Prepare the manager for Slice Machine actions.
+		try {
+			// TODO: This try...catch statement is a temporary fix
```

Comment:
🤔

Reply:
Yeah, I'm not sure what is causing this issue. We have a Linear ticket for this error here: https://linear.app/prismic/issue/SMX-77/puppeteer-crashes-immediately-after-starting-slice-machine

Reply:
Are you able to take screenshots on your machine?